analysis/cohort_joiner.py

```diff
@@ -18,25 +19,13 @@ def read_dataframe(path):
 
 
 def write_dataframe(dataframe, path):
-    # Aside from argument names and types, we use the same code here as we do in
-    # `cohortextractor.pandas_utils.dataframe_to_file`.
-    ext = get_extension(path)
-    if ext == ".csv" or ext == ".csv.gz":
-        dataframe.to_csv(path, index=False)
-    elif ext == ".feather":
-        dataframe.to_feather(path)
-    elif ext == ".dta" or ext == ".dta.gz":
-        dataframe_copy = dataframe.copy(deep=False)
-        for column, dtype in list(dataframe_copy.dtypes.items()):
-            if dtype.name == "category":
-                dataframe_copy[column] = dataframe_copy[column].__array__()
-        convert_dates = {
-            column: "td"
-            for (column, dtype) in dataframe_copy.dtypes.items()
-            if dtype.name == "datetime64[ns]"
-        }
-        dataframe_copy.to_stata(path, write_index=False, convert_dates=convert_dates)
-    else:
+    # We refactored this function, replacing copy-and-paste code from cohort-extractor
+    # with a call to cohort-extractor itself. However, the error types differed. We
+    # preserved the pre-refactoring error type.
+    try:
```

Comment:
This makes perfect sense to me. Have you considered what the next steps will be when `databuilder` comes more online, and we start not to update or maintain `cohortextractor` with the same gusto?

Reply:
That depends on our approach to writing files across Data Builder and cohort-extractor; Stata files specifically, as we modify these the most. We should, arguably, have a utility package. However, the same caveats would apply to a utility package as to other packages: we'd have to update all the images that used it, so we'd have to be sure it was reasonably stable.